### PR TITLE
Recipe retouch for st2web

### PR DESCRIPTION
**Pull request by arm4b**
Address notes missed in https://github.com/StackStorm/chef-stackstorm/pull/45. Finalize work on https://github.com/StackStorm/chef-stackstorm/issues/24

* [x] Generate self_signed certificate by default (conform with default installer logic)
* [x] Add missing Readme notes
* [x] `web` recipe should be self-sufficient (missing packagecloud repo)
* [x] InSpec
* [x] Fix `nginx` service on `Ubuntu Trusty` (upstream bug https://github.com/chef-cookbooks/chef_nginx/pull/55 https://github.com/chef-cookbooks/chef_nginx/issues/57)

### CODE REVIEW

File: test/recipes/default/web_test.rb [hunk 1]
```diff
@@ -5,31 +5,47 @@
...
+describe package('nginx') do
   it { should be_installed }
 end
 
+puts package('nginx').version
+
+describe service('nginx') do
+  it { should be_installed }
+  it { should be_enabled }
+  it { should be_running }
```
Comment (arm4b) [inline test/recipes/default/web_test.rb:19]: Apparently, this reveals a bug (?) in upstream nginx cookbook.
Looks like the service reporting stopped status, but nginx process is running on Ubuntu Trusty. Might be something with `init` files.

Comment (arm4b) [reply test/recipes/default/web_test.rb:19]: Yeah, upstream bug: https://github.com/chef-cookbooks/chef_nginx/pull/55

Comment (shortdudey123) [reply test/recipes/default/web_test.rb:19]: 👍

**Comment by arm4b** [comment]
Related fixes made in upstream `chef-nginx` cookbook: 
https://github.com/chef-cookbooks/chef_nginx/pull/55
https://github.com/chef-cookbooks/chef_nginx/issues/57
https://github.com/chef-cookbooks/chef_nginx/pull/59


CentOS7 CI is failing due to different reason: https://github.com/StackStorm/chef-stackstorm/issues/58 (same for `master`)

**Review by shortdudey123 — APPROVED** [review]

**Comment by shortdudey123** [comment]
I think you can override the merge even though the CI test for centos7 fails since you are an admin

I will see what i can find for https://github.com/StackStorm/chef-stackstorm/issues/58

**Comment by arm4b** [comment]
Merged.

@shortdudey123 Thanks for taking a look!

**Pull request merged**